azure-pipelines.yml

```diff
@@ -0,0 +1,191 @@
+
+        - task: CmdLine@2
+          condition: eq(variables['Build.SourceBranch'], 'refs/release')
+          displayName: 'Publish'
+          inputs:
+            workingDirectory: '$(Build.SourcesDirectory)/device_calendar'
+            script: '$(flutterExecPath) pub publish --force'
+            
+  # ----------- CD Production -----------
+  - stage: Release
+    dependsOn: Build
+    condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/master'))
+    jobs: 
+     - job: Release
+       pool:
+        vmImage: 'windows-latest'
+       steps:
+        - task: FlutterInstall@0
+          displayName: 'Flutter install'
+          inputs:
+            channel: 'stable'
+            version: 'latest'
+
+        - task: PowerShell@2
+          displayName: 'Get version from pubspec.yaml'
+          inputs:
+            targetType: 'inline'
+            script: |
+              [string] $version = select-string -path "$(Build.SourcesDirectory)/device_calendar/pubspec.yaml" -pattern "(?<=version: ).*" | %{ $_.Matches[0].Value }
+              
+              Write-Host "##vso[task.setvariable variable=currentVersion]$version"
+
+        - task: PowerShell@2
+          displayName: 'Get version summary from CHANGELOG.md'
+          inputs:
+            targetType: 'inline'
+            script: |
+              [string] $summary = select-string -path "$(Build.SourcesDirectory)/device_calendar/CHANGELOG.md" -pattern "($([regex]::escape($env:currentVersion))).*" | %{ $_.Matches[0].Value }
+              
+              Write-Host "##vso[task.setvariable variable=versionSummary]$summary"
+
+        - task: GitHubReleasePublish@1
+          displayName: 'Tag a release to GitHub'
+          inputs:
+            githubEndpoint: 'GitHub Tagging'
+            manuallySetRepository: false
+            githubRepository: 'builttoroam/flutter_plugins'
+            githubTag: 'v$(currentVersion)'
+            githubReleaseTitle: '$(versionSummary)'
+            githubReleaseDraft: false
+            githubReleasePrerelease: false
+            githubIgnoreAssets: false
+            githubReleaseAsset: '$(Build.ArtifactStagingDirectory)/*'
+            githubReuseRelease: false
+            githubReuseDraftOnly: false
+            githubSkipDuplicatedAssets: false
+            githubEditRelease: false
+            githubDeleteEmptyTag: false
+
+        - task: CmdLine@2
+          displayName: 'Dry run publish'
+          inputs:
+            workingDirectory: '$(Build.SourcesDirectory)/device_calendar'
+            script: '$(flutterExecPath) pub publish --dry-run'
+
+        - task: CmdLine@2
+          displayName: 'Publish'
+          inputs:
+            workingDirectory: '$(Build.SourcesDirectory)/device_calendar'
+            script: '$(flutterExecPath) pub publish --force'
+
+        - task: GitHubRelease@1
+          displayName: 'Remove GitHub tag when publish fails'
+          condition: failed()
```

Comment:
I don't think it's a big deal but the [`failed()` expression](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops#failed) will evaluate to `True` whenever any previous task will fail, meaning that even when any PowerShell script fails this will run (at least that's my reading of it). It's not ideal and maybe we could find a way to limit this run to only run when the `Publish` or `Dry-Run` task fails? Basically anything after setting up the tag taks

Reply:
You're right and I'm aware of this and couldn't find a way to do that on the task level.

Reply:
I'll have more look but cannot guarantee

Reply:
You might be right that there's no way of doing it (@nickrandolph ?), maybe we should then split this into multiple jobs? That way you could use `dependsOn` with the `condition: failed()`?

Reply:
I tried `dependsOn` but cannot use it for tasks level (I guess `dependsOn` is available for stage level only)

Reply:
You could use `dependsOn` at a job level as well. That was my suggestion to potentially split up these task into different jobs and make them depend on each other. Have a think about this - this is definitely not a blocker for this PR and can be done at later stage.

Reply:
Added a task: https://dev.azure.com/builttoroam/Flutter%20Plugins/_workitems/edit/20528